docs/architecture/adr-028-public-key-addresses.md

```diff
@@ -56,42 +89,90 @@ The current multisig public keys use amino serialization to generate the address
 those public keys and their address formatting, and call them "legacy amino" multisig public keys
 in protobuf. We will also create multisig public keys without amino addresses to be described below.
 
+### Hash Function Choice
+
+We propose to use [blake2b](https://www.blake2.net/) as a hash function choice:
```

Comment:
I'm personally supportive of `blake2b-256` for this purpose, but my impression at the last architecture call was that there will be a lot of opposition... not sure I want to pick this battle... But I guess if others change their mind, like I said I'm willing to support this.

Reply:
Yeah, I will not strongly defend it. If there will be opposition I will just change this section to `sha2`.
We already had lot of discussion about hashing. In general, we would like to have configurable hashing. But this won't work for account generation. We need to have single account generation, because lot tools (wallets, frameworks...) will relay on it.

So we need to select something long term. We won't afford changing it once it will be set.

Reply:
> So we need to select something long term. We won't afford changing it once it will be set.

I strongly disagree with this sentiment. We will need to support using alternate hash functions for addresses unless we want to preclude having provability on addresses, which imo we should expect many chains to want, given what we see in Ethereum.

Reply:
> I will just change this section to sha2.

If I recall correctly, in the last architecture call there was consensus on using _sha2_ for the time being. I feel like setting blake2 here would be perceived as we ignore the general consensus. Thus, I'm against setting `blake2b` as default.

Reply:
@ValarDragon - migrating addresses is pain in the ass because addresses are used in off-chain world as well. 
Also tools, wallets especially, will relay on a simple algorithm. It might be challenging for wallet providers to add customization for every comsmos chain. From user perspective, I would like to have a wallet (and other tools) which will work with all Cosmos SDK chains.

BTW: what do you want with  "_having provability on addresses_"?

Reply:
> I feel like setting blake2 here would be perceived as if we ignore the general consensus.

@alessio, Since then we got new arguments for using blake2b. We also were expressing a need for customizing  a  hash function "globally" for the whole SDK. Not sure if this is what @ValarDragon had in mind.
But if we will not reach a consensus then fine - we will stick with sha2.

Reply:
OK! So til then I'd propose the following:

```suggestion
```

Reply:
> I strongly disagree with this sentiment. We will need to support using alternate hash functions for addresses unless we want to preclude having provability on addresses, which imo we should expect many chains to want, given what we see in Ethereum.

Can you explain a little bit more this use case @ValarDragon ?

This is about a standardized address generation algorithm. Nothing here would preclude any chain from generating addresses differently using a different hash algorithm for any key they wanted. But we're trying to standardize address generation in the SDK so yes I think it should be one hash function.

Reply:
> If I recall correctly, in the last architecture call there was consensus on using sha2 for the time being. 

This was my understanding as well.

Reply:
Can this be updated to `sha2`, as it seems that was the last consensus?

Reply:
updated to sha2